### DISCUSSION TITLE
How to move a tab / buffer to a different position ?

### DISCUSSION THREAD

**Discussion by notDavid** [LazyVim/LazyVim#3100] [Q&A]
Hi there, noob question;

So a very basic example: I open 3 files. So now i have 3 tabs open, each tab displaying a buffer / file.

Lets say:
- tab1 = buffer1 = file1
- **tab2 = buffer2 = file2**
- tab3 = buffer3 = file3

What i want however, is to "swap" buffer1 and buffer2, like so:
- **tab1 = buffer2 = file2**
- tab2 = buffer1 = file 1
- tab3 = buffer3 = file 3

How do i do this?

Thank you!

**Comment by dpetka2001** [comment]
Do you mean actual Neovim tabs? Or what you're visually seeing on the top in the bufferline that mimics Vscode (and other editors)?

**Reply by notDavid** [reply]
Thank you for your reply @dpetka2001 , sorry i should have included a screenshot... :

<details>
  <summary>screenshot</summary>

  ![lazyvim_tabs](https://github.com/LazyVim/LazyVim/assets/1530401/1e9d5f49-893a-4ddb-96b8-8bac1b1f70ea)

</details>

**Reply by dpetka2001** [reply]
I'm on mobile now and can't check but if you type `:` to open command line and then type `bufferline` and press `<Tab>` to get completions it will list all available commands. I believe there's something like `BufferlineMoveLeft` if i'm not mistaken? You should be able to find it.

**Comment by notDavid** [comment]
Great, thank you very much @dpetka2001  🤗  It was `BufferLineMovePrev`.

For future reference:

- `:help bufferline-mappings`


> A few of this plugins commands can be mapped for ease of use.
> These commands will move the current buffer backwards or forwards in the bufferline

    `nnoremap <silent><mymap> :BufferLineMoveNext<CR>`
    `nnoremap <silent><mymap> :BufferLineMovePrev<CR>`

**Comment by Patzi275** [comment]
There are some keybinding for anyone how want:
```
vim.keymap.set("n", "<leader>bH", "<cmd>BufferLineMovePrev<CR>", { desc = "Move buffer left" })
vim.keymap.set("n", "<leader>bL", "<cmd>BufferLineMoveNext<CR>", { desc = "Move buffer right" })
```

**Reply by notDavid** [reply]
@Patzi275 Hi, fyi this was already implemented in LazyVim: https://github.com/LazyVim/LazyVim/pull/1935/files

**Reply by Patzi275** [reply]
> @Patzi275 Hi, fyi this was already implemented in LazyVim: https://github.com/LazyVim/LazyVim/pull/1935/files

Alright, I didn't know, thanks 👌
